# Ultralytics Framework for YOLOv8 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolov08m Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolov8m.pt")

100%|██████████| 49.7M/49.7M [00:00<00:00, 243MB/s]


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 54.8ms
Speed: 6.5ms preprocess, 54.8ms inference, 445.5ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.13 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)


engine/trainer: task=detect, mode=train, model=yolov8m.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, sho

100%|██████████| 6.25M/6.25M [00:00<00:00, 111MB/s]


AMP: checks passed ✅


train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.001), 83 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      12.8G     0.9945      2.223      1.049          5        640: 100%|██████████| 36/36 [00:14<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.06it/s]


                   all       1274       7377      0.627      0.394      0.394      0.262

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      13.1G     0.9088     0.7772     0.9809         17        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.638      0.352      0.317      0.192

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      12.9G     0.9483     0.7882      1.004         37        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.546      0.391      0.321      0.168

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100        13G     0.9343     0.7601      1.008          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.601      0.434      0.417      0.242

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100        13G     0.9101     0.6992     0.9838          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.24it/s]


                   all       1274       7377      0.758      0.461      0.514      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100        13G     0.8765     0.6365      0.968         26        640: 100%|██████████| 36/36 [00:12<00:00,  2.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.736      0.455      0.527      0.333

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      13.1G     0.8674     0.6269     0.9683          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377       0.76      0.513      0.573      0.344

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100        13G     0.8592     0.5925     0.9472         24        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377        0.8       0.53      0.594       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100        13G     0.8285     0.5924     0.9588          1        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.772      0.502      0.567      0.343

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100        13G     0.8255     0.5548     0.9374          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.867      0.536      0.625      0.389

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100        13G     0.8352     0.5506     0.9339         14        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377       0.79      0.545      0.616      0.387

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100        13G     0.8316     0.5965     0.9198          1        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.808      0.556      0.613      0.382

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      13.1G     0.7941     0.5201     0.9302         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.816      0.568      0.633      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      12.9G     0.7889     0.5209      0.922         14        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.24it/s]


                   all       1274       7377      0.835      0.568      0.633      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100        13G     0.7786     0.4974     0.9171         24        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.27it/s]


                   all       1274       7377       0.81       0.59      0.645      0.408

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      13.1G     0.7718     0.4978     0.9247          4        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.839      0.568      0.639        0.4

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      13.1G     0.7911      0.501     0.9407         15        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.832      0.576      0.651      0.404

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      12.9G     0.7738     0.5026     0.9358          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.847      0.587      0.656      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      13.1G     0.7684     0.4788     0.9174         12        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.26it/s]


                   all       1274       7377      0.835      0.585      0.658      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      13.1G     0.7557     0.4727     0.9179         23        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.857      0.589      0.673      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      13.1G     0.7354     0.4552     0.9112          4        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.855        0.6      0.675      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100        13G     0.7724     0.4861     0.9173          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.844        0.6      0.672      0.431

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100        13G     0.7242     0.4584     0.9051         27        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.819      0.591       0.66      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100        13G     0.7207     0.4416     0.9051          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.859      0.604      0.684      0.442

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100        13G     0.7235     0.4499     0.9094         20        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.863      0.606      0.696      0.443

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      12.9G     0.7127     0.4436     0.9174         18        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.842      0.611      0.687      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      13.1G     0.7241     0.4453     0.9015         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.846      0.617      0.684      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100        13G     0.7069     0.4336     0.8973         20        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.19it/s]


                   all       1274       7377      0.866      0.611      0.683      0.441

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      13.1G     0.6808     0.4146     0.8894          7        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.889      0.616      0.699      0.455

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100        13G     0.7119     0.4309     0.9019         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.887       0.61      0.696      0.444

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100        13G     0.6818      0.417     0.8937         19        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.16it/s]


                   all       1274       7377      0.858      0.627        0.7      0.449

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      13.1G     0.6988     0.4223     0.8919         10        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.867      0.611      0.692      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100        13G     0.6792     0.4138     0.8907         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.27it/s]


                   all       1274       7377      0.873      0.616      0.704      0.458

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      13.1G     0.6936     0.4153     0.9041         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.27it/s]


                   all       1274       7377      0.858      0.632        0.7      0.452

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      13.1G      0.697      0.417     0.9032         21        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.882       0.62      0.702       0.46

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      13.1G     0.6749     0.4066     0.8902          9        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.26it/s]


                   all       1274       7377      0.867      0.631      0.707      0.461

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      12.9G     0.6844     0.4128     0.8986         10        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.876      0.624      0.708      0.467

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      12.9G      0.659     0.4124     0.8734          0        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.862      0.637      0.709      0.464

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100        13G     0.6627     0.3888     0.8886         21        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.859      0.643      0.716      0.465

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100        13G     0.6662     0.3939     0.8862         15        640: 100%|██████████| 36/36 [00:12<00:00,  2.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.859      0.634      0.718      0.467

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      12.9G     0.6564     0.3997     0.8915         10        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.22it/s]


                   all       1274       7377      0.859      0.631      0.712      0.466

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100        13G     0.6598      0.398     0.8863         24        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.10it/s]


                   all       1274       7377      0.877      0.633      0.715      0.464

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100        13G     0.6514     0.4261     0.8908          4        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.06it/s]


                   all       1274       7377      0.882      0.628      0.715      0.479

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100        13G     0.6666     0.3837     0.8757         26        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.26it/s]


                   all       1274       7377      0.866      0.651      0.729      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100        13G     0.6372     0.3735     0.8884          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.858      0.652      0.728      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      13.1G     0.6271     0.3678     0.8837          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.877      0.636      0.725      0.479

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      12.9G     0.6383      0.374     0.8795         15        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.862      0.651      0.726      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      12.9G     0.6511     0.3726      0.881         16        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.879      0.639      0.727      0.479

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      12.9G     0.6721      0.493     0.8647          7        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.871      0.652      0.728      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100        13G     0.6488     0.3703     0.8752         17        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.881      0.643      0.723      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      13.1G     0.6275     0.3665      0.886         10        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.867      0.644      0.726      0.487

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100        13G     0.6197     0.3587     0.8788          5        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.858      0.662      0.739      0.486

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      12.9G     0.6161     0.3601     0.8794          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.14it/s]


                   all       1274       7377      0.875      0.651      0.734      0.487

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100        13G     0.6075     0.3548     0.8736         14        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.887      0.643      0.733      0.488

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      13.1G     0.6203     0.3645     0.8748         21        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.26it/s]


                   all       1274       7377      0.862       0.66      0.736       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100        13G     0.5995     0.3486      0.877          5        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377       0.87       0.66       0.74      0.493

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      13.1G      0.621     0.3569     0.8664         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.869      0.667      0.743      0.496

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100        13G     0.6051     0.3492     0.8758         21        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.22it/s]


                   all       1274       7377      0.884      0.645      0.734      0.487

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100        13G     0.6141     0.3466     0.8713         12        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.892      0.646       0.74      0.491

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      12.9G     0.6066     0.3428     0.8741         18        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.12it/s]


                   all       1274       7377      0.889      0.655      0.748      0.497

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100        13G     0.6086     0.3469     0.8688         39        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.876       0.66      0.747      0.499

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100        13G     0.5968     0.3367     0.8739         33        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377       0.89       0.66      0.744      0.496

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100        13G     0.6047     0.3418      0.866         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.23it/s]


                   all       1274       7377      0.895      0.652       0.75      0.503

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100        13G     0.6151     0.3451     0.8643          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  2.95it/s]


                   all       1274       7377      0.867      0.666       0.75      0.504

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      13.1G     0.6208     0.3383      0.868          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.25it/s]


                   all       1274       7377       0.88      0.661      0.747      0.501

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100        13G     0.5959     0.3371     0.8687         10        640: 100%|██████████| 36/36 [00:13<00:00,  2.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.21it/s]


                   all       1274       7377      0.881      0.674      0.753      0.502

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      13.1G     0.6045     0.3426     0.8685         19        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.875       0.67       0.76      0.509

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100        13G      0.578     0.3296     0.8583         19        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.887      0.657      0.747      0.503

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      12.9G     0.6011     0.3374     0.8658         16        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.26it/s]


                   all       1274       7377      0.879      0.667      0.754      0.504

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      12.9G     0.5871     0.3311      0.866         24        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.886      0.667      0.754      0.507

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      13.1G     0.5791     0.3251     0.8627         17        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.891      0.668      0.753      0.506

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100        13G     0.5815     0.3228     0.8638         12        640: 100%|██████████| 36/36 [00:12<00:00,  2.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.872      0.668      0.762      0.514

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100        13G     0.5788      0.329      0.871         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.892      0.662      0.757      0.508

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      12.9G     0.5658      0.322     0.8702         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.892      0.664      0.754      0.506

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      13.1G     0.5902     0.3201      0.872          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.28it/s]


                   all       1274       7377      0.887      0.663       0.76      0.512

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      13.1G     0.5808     0.3176     0.8566          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.894      0.667      0.761      0.513

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100        13G     0.5812      0.321     0.8611         22        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.882      0.664      0.753      0.507

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100        13G     0.5761     0.3133     0.8678          4        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.883      0.672       0.76      0.512

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100        13G      0.571     0.3135      0.865         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.889      0.673      0.763      0.517

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      13.1G     0.5594       0.31     0.8574         36        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.881      0.676      0.765      0.517

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100        13G     0.5605     0.3142     0.8597          7        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.11it/s]


                   all       1274       7377      0.871      0.683      0.765      0.514

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      12.9G     0.5688     0.3135     0.8601          9        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.857      0.688      0.762      0.515

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      13.1G     0.5682     0.3089     0.8601          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.861      0.683      0.767      0.519

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100        13G     0.5643     0.3138     0.8594          8        640: 100%|██████████| 36/36 [00:12<00:00,  2.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.18it/s]


                   all       1274       7377      0.888      0.671      0.765      0.517

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100        13G     0.5601     0.3055     0.8656         11        640: 100%|██████████| 36/36 [00:12<00:00,  2.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.893      0.667      0.761      0.517

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      12.9G     0.5978     0.4714     0.8994          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.901      0.657      0.763      0.516

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      12.9G     0.5636     0.3011     0.8605         14        640: 100%|██████████| 36/36 [00:12<00:00,  2.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.27it/s]


                   all       1274       7377      0.876      0.682      0.765      0.519

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      13.1G     0.5541      0.301     0.8565         16        640: 100%|██████████| 36/36 [00:12<00:00,  2.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.35it/s]


                   all       1274       7377      0.881      0.671      0.764      0.519

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100        13G     0.5241     0.2846     0.8588         14        640: 100%|██████████| 36/36 [00:12<00:00,  2.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377       0.88      0.675      0.767       0.52

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100        13G     0.5301     0.2968     0.8648          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.883      0.674      0.766      0.521
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      13.1G     0.6009     0.3022     0.8464         13        640: 100%|██████████| 36/36 [00:13<00:00,  2.70it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.884      0.672      0.761      0.516

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100        13G      0.598      0.301     0.8452         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.895      0.668      0.764       0.52

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      12.9G     0.5899     0.2986     0.8417          7        640: 100%|██████████| 36/36 [00:12<00:00,  2.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.883      0.676      0.766       0.52

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      13.1G     0.6002     0.3038     0.8434          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.31it/s]


                   all       1274       7377      0.886      0.679      0.763      0.521

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      13.1G     0.5997     0.2994     0.8409         31        640: 100%|██████████| 36/36 [00:12<00:00,  2.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.30it/s]


                   all       1274       7377      0.896      0.673      0.767       0.52

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      12.9G     0.5867     0.3024     0.8431         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.34it/s]


                   all       1274       7377      0.887      0.674      0.772      0.524

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      12.9G     0.5889     0.2956     0.8413         13        640: 100%|██████████| 36/36 [00:12<00:00,  2.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377        0.9      0.669      0.772      0.525

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      12.9G     0.5898     0.2943     0.8322          3        640: 100%|██████████| 36/36 [00:12<00:00,  2.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.33it/s]


                   all       1274       7377      0.896      0.671      0.774      0.525

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      12.9G     0.5798     0.2881     0.8364         40        640: 100%|██████████| 36/36 [00:12<00:00,  2.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.29it/s]


                   all       1274       7377      0.885      0.674      0.772      0.525

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100        13G     0.5789     0.2851     0.8401          6        640: 100%|██████████| 36/36 [00:12<00:00,  2.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  3.32it/s]


                   all       1274       7377      0.892      0.672      0.771      0.526

100 epochs completed in 0.816 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/last.pt, 52.0MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt, 52.0MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.21it/s]


                   all       1274       7377      0.892      0.672      0.771      0.525
            Pedestrian        333        600      0.833       0.72      0.785      0.478
         Traffic_Signs        309        434      0.893      0.618      0.728      0.453
               Vehicle       1034       2820       0.94       0.89      0.946      0.778
         traffic_light        660       3523      0.901      0.461      0.626      0.394
Speed: 0.1ms preprocess, 2.4ms inference, 0.0ms loss, 0.8ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train


# Results on Validation set:
100 epochs completed in 0.816 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/last.pt, 52.0MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt, 52.0MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:09<00:00,  2.21it/s]
                   all       1274       7377      0.892      0.672      0.771      0.525
            Pedestrian        333        600      0.833       0.72      0.785      0.478
         Traffic_Signs        309        434      0.893      0.618      0.728      0.453
               Vehicle       1034       2820       0.94       0.89      0.946      0.778
         traffic_light        660       3523      0.901      0.461      0.626      0.394
Speed: 0.1ms preprocess, 2.4ms inference, 0.0ms loss, 0.8ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.72s/it]


                   all        641       3701      0.886       0.67      0.802      0.588
            Pedestrian        182        332      0.783      0.705       0.79      0.536
         Traffic_Signs        168        230      0.923      0.626      0.792      0.529
               Vehicle        528       1377      0.948      0.895      0.942      0.815
         traffic_light        338       1762      0.891      0.453      0.684       0.47
Speed: 0.3ms preprocess, 4.8ms inference, 0.0ms loss, 1.9ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.72s/it]
                   all        641       3701      0.886       0.67      0.802      0.588
            Pedestrian        182        332      0.783      0.705       0.79      0.536
         Traffic_Signs        168        230      0.923      0.626      0.792      0.529
               Vehicle        528       1377      0.948      0.895      0.942      0.815
         traffic_light        338       1762      0.891      0.453      0.684       0.47
Speed: 0.3ms preprocess, 4.8ms inference, 0.0ms loss, 1.9ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv8n Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.51s/it]


                   all        641       3701      0.886       0.67      0.802      0.588
            Pedestrian        182        332      0.783      0.705       0.79      0.536
         Traffic_Signs        168        230      0.923      0.626      0.792      0.529
               Vehicle        528       1377      0.948      0.895      0.942      0.815
         traffic_light        338       1762      0.891      0.453      0.684       0.47
Speed: 0.3ms preprocess, 4.6ms inference, 0.0ms loss, 2.5ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8m - 0.315
- IOU threshold : 0.5
- Results save d at: runs/detect/val3

In [2]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.315, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.41s/it]


                   all        641       3701      0.907      0.652      0.798      0.589
            Pedestrian        182        332      0.815      0.678      0.785      0.538
         Traffic_Signs        168        230      0.939      0.604      0.785      0.529
               Vehicle        528       1377      0.963      0.887      0.939      0.815
         traffic_light        338       1762      0.911      0.438      0.683      0.475
Speed: 1.7ms preprocess, 5.0ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best + Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8m - 0.315
- IOU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.315, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.48s/it]


                   all        641       3701      0.907      0.652      0.798      0.589
            Pedestrian        182        332      0.815      0.678      0.785      0.538
         Traffic_Signs        168        230      0.939      0.604      0.785      0.529
               Vehicle        528       1377      0.963      0.887      0.939      0.815
         traffic_light        338       1762      0.911      0.438      0.683      0.475
Speed: 1.9ms preprocess, 4.7ms inference, 0.0ms loss, 1.5ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best +  Optimal value + IoU 0.6
- Optimal Confidence YOLOv8m - 0.315
- IOU threshold : 0.6
- Results save d at: runs/detect/val5

In [1]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.315, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.40s/it]


                   all        641       3701      0.906      0.652      0.798      0.589
            Pedestrian        182        332      0.812      0.678      0.784      0.538
         Traffic_Signs        168        230      0.939      0.604      0.785      0.529
               Vehicle        528       1377      0.961      0.888      0.939      0.815
         traffic_light        338       1762       0.91      0.438      0.682      0.475
Speed: 1.9ms preprocess, 4.8ms inference, 0.0ms loss, 1.6ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


In [2]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.315, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.75, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.51s/it]


                   all        641       3701      0.895      0.653      0.796      0.588
            Pedestrian        182        332      0.796      0.681      0.782      0.536
         Traffic_Signs        168        230      0.933      0.604      0.784      0.529
               Vehicle        528       1377      0.947      0.888      0.938      0.814
         traffic_light        338       1762      0.905      0.439      0.681      0.473
Speed: 2.0ms preprocess, 4.2ms inference, 0.0ms loss, 0.4ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


In [3]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.315, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.95, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:49<00:00,  8.28s/it]


                   all        641       3701      0.711      0.625      0.713      0.533
            Pedestrian        182        332      0.538      0.633      0.624      0.437
         Traffic_Signs        168        230      0.805      0.575      0.728      0.492
               Vehicle        528       1377      0.768      0.878      0.905      0.786
         traffic_light        338       1762      0.733      0.413      0.595      0.416
Speed: 2.0ms preprocess, 4.2ms inference, 0.0ms loss, 0.6ms postprocess per image
Saving runs/detect/val7/predictions.json...
Results saved to runs/detect/val7


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_v8/v8_m/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 218 layers, 25,842,076 parameters, 0 gradients, 78.7 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:35<00:00,  1.01it/s]


                   all       4485      26788      0.914      0.712      0.837       0.66
            Pedestrian       1229       2227      0.862      0.788      0.869      0.657
         Traffic_Signs       1093       1459      0.907      0.682       0.82      0.599
               Vehicle       3636      10031      0.963      0.924      0.959      0.861
         traffic_light       2373      13071      0.925      0.456      0.701      0.523
Speed: 0.4ms preprocess, 4.3ms inference, 0.0ms loss, 0.6ms postprocess per image
Saving runs/detect/val8/predictions.json...
Results saved to runs/detect/val8


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.